src/components/product-list/useProductAsync.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+
+export const useProductAsync = () => {
+  const [products, setProducts] = useState([]);
+  const [loader, setLoader] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        setLoader(true);
+        const { data } = await axios.get('/api/products');
```

Comment:
As you've extracted the code in separate folders, the same thing could be done with network requests. The network request could be extracted out in 'services' folder in a services file. The file should be named `name.service.js`. e.g. `getProducts.service.js`